inc/Engine/Optimization/RUCSS/Frontend/Subscriber.php

```diff
@@ -0,0 +1,158 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Engine\Optimization\RUCSS\Frontend;
+
+use WP_Rocket\Logger\Logger;
+use WP_Rocket\Admin\Options_Data;
+use WP_Rocket\Event_Management\Subscriber_Interface;
+use WP_Rocket\Engine\Optimization\RUCSS\APIClient;
+use WP_Rocket\Engine\Optimization\RUCSS\Controller\UsedCSS;
+
+class Subscriber implements Subscriber_Interface {
+	/**
+	 * Plugin options instance.
+	 *
+	 * @var Options_Data
+	 */
+	protected $options;
+
+	/**
+	 * APIClient instance
+	 *
+	 * @var APIClient
+	 */
+	private $api;
+
+	/**
+	 * UsedCss instance
+	 *
+	 * @var UsedCSS
+	 */
+	private $used_css;
+
+	/**
+	 * Instantiate the class
+	 *
+	 * @param Options_Data $options      Plugin options instance.
+	 * @param UsedCSS      $used_css Settings instance.
+	 * @param APIClient    $api      Database instance.
+	 */
+	public function __construct( Options_Data $options, UsedCSS $used_css, APIClient $api ) {
+		$this->options  = $options;
+		$this->used_css = $used_css;
+		$this->api      = $api;
+	}
+
+	/**
+	 * Return an array of events that this subscriber listens to.
+	 *
+	 * @return array
+	 */
+	public static function get_subscribed_events() : array {
+		return [
+			'rocket_buffer' => [ 'treeshake', 1 ],
+		];
+	}
+
+	/**
+	 * Apply TreeShaked CSS to the current HTML page.
+	 *
+	 * @param string $html  HTML content.
+	 *
+	 * @return string  HTML content.
+	 */
+	public function treeshake( string $html ) : string {
+		if ( ! $this->is_allowed() ) {
+			return $html;
+		}
+
+		global $wp;
+		$url       = untrailingslashit( home_url( add_query_arg( [], $wp->request ) ) );
+		$is_mobile = $this->is_mobile();
+		$used_css  = $this->used_css->get_used_css( $url, $is_mobile );
+
+		if ( empty( $used_css ) || ( $used_css->retries < 3 ) ) {
+			$customer_key = ! empty( $this->options->get( 'consumer_key', '' ) )
+				? $this->options->get( 'consumer_key', '' )
+				: rocket_get_constant( 'WP_ROCKET_KEY', '' );
```

Comment:
The constant check is already done in the `Options_Data` class, so you don't need to re-do it here